widget/entry.go

```diff
@@ -297,12 +486,122 @@ func (e *Entry) TypedRune(r rune) {
 	Renderer(e).(*entryRenderer).moveCursor()
 }
 
+// KeyDown handler for keypress events - used to store shift modifier state for text selection
+func (e *Entry) KeyDown(key *fyne.KeyEvent) {
+	// For keyboard cursor controlled selection we now need to store shift key state and selection "start"
+	// Note: selection start is where the highlight started (if the user moves the selection up or left then
+	// the selectRow/Column will not match SelectionStart)
+	if key.Name == desktop.KeyShiftLeft || key.Name == desktop.KeyShiftRight {
+		if e.selecting == false {
+			e.selectRow = e.CursorRow
+			e.selectColumn = e.CursorColumn
+		}
+		e.selectKeyDown = true
+	}
+}
+
+// KeyUp handler for key release events - used to reset shift modifier state for text selection
+func (e *Entry) KeyUp(key *fyne.KeyEvent) {
+	// Handle shift release for keyboard selection
+	// Note: if shift is released then the user may repress it without moving to adjust their old selection
+	if key.Name == desktop.KeyShiftLeft || key.Name == desktop.KeyShiftRight {
+		e.selectKeyDown = false
+	} else {
+		if e.selectKeyDown == false {
+			e.selecting = false
+		}
+	}
+}
+
+// eraseSelection removes the current selected region and moves the cursor
+func (e *Entry) eraseSelection() {
+	if e.ReadOnly {
+		return
+	}
+
+	provider := e.textProvider()
+	posA := e.SelectionStart()
+	posB := e.SelectionEnd()
+
+	if posA == posB {
+		return
+	}
+
+	e.Lock()
+
+	provider.deleteFromTo(posA, posB)
+	e.CursorRow, e.CursorColumn = e.rowColFromTextPos(posA)
+	e.selectRow, e.selectColumn = e.CursorRow, e.CursorColumn
+	e.Unlock()
+	e.selecting = false
+}
+
 // TypedKey receives key input events when the Entry widget is focused.
 func (e *Entry) TypedKey(key *fyne.KeyEvent) {
 	if e.ReadOnly {
 		return
 	}
+
+	// seeks to the start/end of the selection - used by: up, down, left, right
+	seekSelection := func(start bool) {
+		var pos int
+		if start {
+			pos = e.SelectionStart()
+		} else {
+			pos = e.SelectionEnd()
+		}
+		e.Lock()
+		e.CursorRow, e.CursorColumn = e.rowColFromTextPos(pos)
+		e.Unlock()
+		e.selecting = false
```

Comment:
Why is a seek function terminating the selection?

Reply:
seekSelection visibly moves the cursor to a new location in the document. Technically this action shouldn't disable selection but, in practice, every caller using seekSelection needs to disable selection in order to provide familiar usability to other text editors.

Reply:
I see that. If it is intended behaviour then the method name should indicate it I think, otherwise move that out (it's only called in 2 places I think?)